src/pages/postgraphile/usage-schema.md

```diff
@@ -238,3 +238,288 @@ look at the `postgraphile-core` and `graphile-build-pg` modules.
+   section.
+5. Create a file `codegen.yml` in the root of your workspace.
+   ```yaml
+   overwrite: true
+   schema: "./src/generated/schema.graphql"
+   generates:
+     # Creates the TypeScript types from the schema and any .graphql file
+     src/generated/types.ts:
+       documents: "src/**/*.graphql"
+       plugins:
+         - typescript
+         - typescript-operations
+         - typescript-urql
+       config:
+         withHOC: false
+         withComponent: false
+         withMutationFn: false
+   config:
+     scalars:
+       DateTime: "string"
+       JSON: "{ [key: string]: any }"
+   ```
+6. Run `graphql-codegen --config codegen.yml` to generate the types.
+7. The generated types can now be used in your custom business logic code.
+
+### Example
+
+We have a movie table that we want to query from our backend system.
+
+We can write a small GraphQL query file similar to this. It could be stored in
+`./src/graphql/getMovies.graphql`.
+
+```graphql
+query GetMovies($top: Int!) {
+  movies(first: $top) {
+    nodes {
+      id
+      title
+    }
+  }
+}
+```
+
+Save the file and run the code generation task.
+
+Alternatively we can create some inline query directly in code like this:
+
+```js
+import { makeExtendSchemaPlugin, gql } from "graphile-utils";
+
+// inside some function:
+
+const GetMoviesDocument = gql`
+  query Query {
+    __typename
+    movies(first: 3) {
+      nodes {
+        id
+        title
+      }
+    }
+  }
+`;
+```
+
+The query can then be used in your code via the generated type or inline:
+
+```js
+
+  // This would be generated from the above .graphql file by the code generator
+  // alternatively define it inline via gql` like shown above
+  const doc = GetMoviesDocument;
+  const variables: GetMoviesQueryVariables = {
+    top: 3,
+  };
+
+  // Execute the query from your backend code
+  // The "Query" type is your the root Query type. It contains all the query
+  // endpoints from the GraphQL API with type support
+  const queryResult = await graphql.execute<Query>(
+    resolveInfo.schema,
+    doc,
+    undefined,
+    context,
+    variables,
+  );
+
+  if (queryResult.errors) {
+    // do something in error case
+  }
+
+  // the result can then be used to get the returned data
+  const allTitles = queryResult.data?.movies?.nodes.map(
+    (movie: any) => movie.title,
+  );
+```
+
+### Usage in different resolvers
+
+To use the GraphQL inside of our backend we need to get access to the "graphql"
+variable. This variable is available for example in the following places:
+
+```js
+export const BusinessLogicPlugin = makeExtendSchemaPlugin((build: Build) => {
+  const { graphql } = build;
+```
+
+```js
+export const WrapResolverPlugin = makeWrapResolversPlugin({
+  Mutation: {
+    async updateMovieById(
+      resolve: any,
+      source: any,
+      args: { [argName: string]: any },
+      context: any,
+      resolveInfo: GraphQLResolveInfo,
+    ) {
+    const query = context.pgClient.query;
```

Comment:
This doesn't show where to get `build.graphql` from... Also I'm not sure if it's possible to get `build` from Method 1 `makeWrapResolversPlugin` currently :flushed:

Reply:
Oh... I think I mixed that one with the PostgreSQL client. `build` is indeed not available. I think this would be a good addition to the context object to add the `graphql` object (in addition to the `pgClient`). For now, I will remove that section as the makeExtendSchemaPlugin is contained in the above example now.